### DISCUSSION TITLE
user_data_dir param to save browser profile not working

### DISCUSSION THREAD

**Discussion by caffenie** [seleniumbase/SeleniumBase#2775] [Q&A] [answered]
sorry if this is a newbie question, but I have been trying for hours to get browser profiles working to no avail. every time I add the user_data_dir parameter to SB (path points to an empty folder) and launch SB, it opens my default personal chrome (which SB can't control) instead of the chromedriver controlled one. how do I save and run a browser profile?

How can I get browser profiles working? I want to be able to save cookies. As an alternative if this doesn't work, what is the most "stealthy" way to add cookies? I have been told loading the url then adding them is less preferable over adding them to the browser first then loading the url but I am not sure

user_data_dir = "../../data/browser_profiles/prof1"
with SB (user_data_dir =user_data_dir)as sb:
       login()
      sb.driver.quit()

**Comment by mdmintz** [comment, marked as answer]
Duplicate of https://github.com/seleniumbase/SeleniumBase/discussions/2331#discussioncomment-7730581

If you need multiple profiles, set a different `user_data_dir` for each one that you want. Only the `Default` profile is supported.

If you're setting the `user_data_dir` while using UC Mode, then make sure you read https://github.com/seleniumbase/SeleniumBase/discussions/2118#discussioncomment-7029201 first. (Don't mix a regular `user_data_dir` with one that is used by UC Mode Chrome, as the two formats are not compatible.)

**Reply by caffenie** [reply]
Thank you for the reply!
How can I save a UC mode browser profile? 

in selenium user_data_dir = empty folder creates a new one, but this isn't an option as default chrome is launched with an empty folder set to user_data_dir

**Reply by mdmintz** [reply]
Unless manually deleted, any custom `user_data_dir` created will still be there after the UC Mode test completes. Then be sure to specify the same `user_data_dir` again.

(Note that you can configure the `user_data_dir`, but not the Profile directory inside that, which is separate. Only the `Default` profile is used, and that is automatically selected when specifying the `user_data_dir`.)

**Comment by rbrisita** [comment]
The `user_data_dir` should be the full path to avoid Chrome access error.
